android/src/main/java/com/swmansion/reanimated/keyboard/KeyboardAnimationManager.java

```diff
@@ -14,21 +15,18 @@ public class KeyboardAnimationManager {
   private final ConcurrentHashMap<Integer, KeyboardWorkletWrapper> mListeners =
       new ConcurrentHashMap<>();
   private final Keyboard mKeyboard = new Keyboard();
-  private final WindowsInsetsManager mWindowsInsetsManager;
+  private final ModalActivityManager mModalActivityManager;
 
   public KeyboardAnimationManager(WeakReference<ReactApplicationContext> reactContext) {
-    mWindowsInsetsManager =
-        new WindowsInsetsManager(reactContext, mKeyboard, this::notifyAboutKeyboardChange);
+    mModalActivityManager =
+        new ModalActivityManager(reactContext, mKeyboard, this::notifyAboutKeyboardChange);
```

Comment:
Can we have two manager here? One for main application window and one for modals?

Reply:
Yeah, that makes the code more straightforward, thanks! 🎉 Fixed in https://github.com/software-mansion/react-native-reanimated/pull/5916/commits/25c1a9439d6e2845654e82dcadbb7da0baede236